sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java

```diff
@@ -359,12 +361,70 @@ private static DefaultResourceSet from(
             resourceSetBuilder.memory(Double.valueOf(memory));
         }
 
-        if (rawEndpoints != null) {
-            resourceSetBuilder.addPorts(rawEndpoints);
+        if (rawPorts != null) {
+            resourceSetBuilder.addResource(from(role, principal, rawPorts));
         }
 
         return resourceSetBuilder
                 .id(id)
                 .build();
     }
+
+    private static ResourceSpec from(String role, String principal, WriteOnceLinkedHashMap<String, RawPort> rawPorts) {
+        Collection<PortSpec> portSpecs = new ArrayList<>();
+        Protos.Value.Builder portsValueBuilder = Protos.Value.newBuilder().setType(Protos.Value.Type.RANGES);
+        String envKey = null;
+        for (Map.Entry<String, RawPort> portEntry : rawPorts.entrySet()) {
+            String name = portEntry.getKey();
+            RawPort rawPort = portEntry.getValue();
+            Protos.Value.Builder portValueBuilder = Protos.Value.newBuilder()
+                    .setType(Protos.Value.Type.RANGES);
+            portValueBuilder.getRangesBuilder().addRangeBuilder()
+                    .setBegin(rawPort.getPort())
+                    .setEnd(rawPort.getPort());
+            portsValueBuilder.mergeRanges(portValueBuilder.getRanges());
+            if (envKey == null) {
+                envKey = rawPort.getEnvKey();
+            }
+
+            if (rawPort.getVip() != null) {
+                final RawVip rawVip = rawPort.getVip();
+                final String protocol =
+                        StringUtils.isEmpty(rawVip.getProtocol()) ? DEFAULT_VIP_PROTOCOL : rawVip.getProtocol();
+                final String vipName = StringUtils.isEmpty(rawVip.getPrefix()) ? name : rawVip.getPrefix();
+                portSpecs.add(new NamedVIPSpec(
+                        Constants.PORTS_RESOURCE_TYPE,
+                        portValueBuilder.build(),
+                        role,
+                        principal,
+                        rawPort.getEnvKey(),
+                        name,
+                        protocol,
+                        toVisibility(rawVip.isAdvertised()),
```

Comment:
It's intentionally undocumented right now for that reason. At the moment all it does is hide the VIP in the endpoints endpoint when set to false (default is true). It'd probably make sense to migrate it to the port level in the next PR, when non-VIPed ports will also be listed in that endpoint. Or just remove it until something actually needs it..